sodapy/__init__.py

```diff
@@ -305,8 +338,35 @@ def _raise_for_status(response):
 
 
 def _clear_empty_values(args):
+    '''
+    Scrap junk data from a dict.
+    '''
     result = {}
     for param in args:
         if args[param] is not None:
             result[param] = args[param]
     return result
+
+def authentication_validation(username, password, access_token):
+    '''
+    Only accept one form of authentication.
+    '''
+    if bool(username) != bool(password):
```

Comment:
This is a quirk of python. I guess in this case I am actually comparing singleton values, so `is not` would be more appropriate than `!=` according to PEP8. I guess the best thing would probably be to simulate a nand by employing some kind of `not ( x and y)`. Will update.